tests/basic/basic_test.go

```diff
@@ -92,21 +94,29 @@ var _ = Describe("{VolumeDriverDownAttachedNode}", func() {
 			contexts = append(contexts, ScheduleAndValidate(fmt.Sprintf("voldriverdownattachednode-%d", i))...)
 		}
 
-		Step("get nodes for all apps in test and restart volume driver", func() {
+		Step("get nodes where app is running and restart volume driver", func() {
 			for _, ctx := range contexts {
-				appNodes := GetNodesThatCanBeDown(ctx)
-				Step(fmt.Sprintf("stop volume driver %s on app %s's nodes: %v",
-					Inst().V.String(), ctx.App.Key, appNodes), func() {
-					StopVolDriverAndWait(appNodes)
-				})
+				appNodes, err := Inst().S.GetNodesForApp(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				for _, appNode := range appNodes {
+					Step(
+						fmt.Sprintf("stop volume driver %s on app %s's node: %s",
+							Inst().V.String(), ctx.App.Key, appNode.Name),
+						func() {
+							StopVolDriverAndWait([]node.Node{appNode})
+						})
 
-				Step("starting volume driver", func() {
-					StartVolDriverAndWait(appNodes)
-				})
+					Step(
+						fmt.Sprintf("starting volume %s driver on app %s's node %s",
+							Inst().V.String(), ctx.App.Key, appNode.Name),
+						func() {
+							StartVolDriverAndWait([]node.Node{appNode})
+						})
 
-				Step("Giving few seconds for volume driver to stabilize", func() {
-					time.Sleep(20 * time.Second)
-				})
+					Step("Giving few seconds for volume driver to stabilize", func() {
+						time.Sleep(20 * time.Second)
+					})
+				}
```

Comment:
validate `ctx` here. Similar as above test. After bouncing PX on each `appNode`, validate the current ctx.

Reply:
The test seems be using `ValidateAndDestroy` at below. Instead move the validate up here and just do destroy in the end.

This should help us catch errors much earlier.

we need to run this in a private job as the test behavior has changed.